parsel/xpathfuncs.py

```diff
@@ -48,6 +49,7 @@ def has_class(context, *classes):
     if node_cls is None:
         return False
     node_cls = ' ' + node_cls + ' '
+    node_cls = re.sub('\s+', ' ', node_cls)
```

Comment:
If you want to be super-strict, `\s+` is not a correct regexp, as it includes unicode whitespaces; css class names are [space separated tokens](https://www.w3.org/TR/html5/infrastructure.html#set-of-space-separated-tokens), where space characters are [more limited](https://www.w3.org/TR/html5/infrastructure.html#space-characters) than what `\s` matches. 

It is also not the same as Python's `\s` with ASCII flag on, as it doesn't include `'\v'`. 

Whitespaces defined in html spec can be found in w3lib: https://github.com/scrapy/w3lib/blob/c1a030582ec30423c40215fcd159bc951c851ed7/w3lib/html.py#L20. As parsel already depends on w3lib, it can be fine to import it from there, though required w3lib version should be bumped.